src/common/libschedutil/ops.c

```diff
@@ -197,49 +212,55 @@ static void service_unregister (flux_t *h)
     flux_future_destroy (f);
 }
 
-struct ops_context *schedutil_ops_register (flux_t *h,
-                                            op_alloc_f *alloc_cb,
-                                            op_free_f *free_cb,
-                                            op_exception_f *exception_cb,
-                                            void *arg)
+int schedutil_ops_register (schedutil_t *s,
+                            op_alloc_f *alloc_cb,
+                            op_free_f *free_cb,
+                            op_exception_f *exception_cb,
+                            void *arg)
 {
     struct ops_context *ctx;
 
-    if (!h || !alloc_cb || !free_cb || !exception_cb) {
+    if (!s || !alloc_cb || !free_cb || !exception_cb) {
         errno = EINVAL;
-        return NULL;
+        return -1;
     }
     if (!(ctx = calloc (1, sizeof (*ctx))))
-        return NULL;
-    if (service_register (h) < 0)
+        return -1;
+    if (service_register (s->h) < 0)
         goto error;
-    ctx->h = h;
     ctx->alloc_cb = alloc_cb;
     ctx->free_cb = free_cb;
     ctx->exception_cb = exception_cb;
     ctx->arg = arg;
-
-    if (flux_msg_handler_addvec (h, htab, ctx, &ctx->handlers) < 0)
+    s->ops_ctx = ctx;
+    if (flux_msg_handler_addvec (s->h, htab, s, &ctx->handlers) < 0)
         goto error;
-    if (flux_event_subscribe (h, "job-exception") < 0)
+    if (flux_event_subscribe (s->h, "job-exception") < 0)
         goto error;
-    return ctx;
+    return 0;
 error:
-    schedutil_ops_unregister (ctx);
-    return NULL;
+    schedutil_ops_unregister (s);
+    return -1;
 }
 
 
-void schedutil_ops_unregister (struct ops_context *ctx)
+void schedutil_ops_unregister (schedutil_t *s)
 {
+    if (!s) {
+        // TODO: how to handle this?
```

Comment:
Generally in a void-returning destructor, we make it a no-op if something is unset.